docroot/modules/custom/mass_utility/src/Commands/MassUtilityCommands.php

```diff
@@ -398,4 +399,95 @@ public function queueRevisionsCleanup(array $options = [
 
   }
 
+  /**
+   * Copy D2D nodes into redirect entities.
+   *
+   * @command copyd2d
+   * @option limit
+   * @option min-hits
+   */
+  public function copyD2DRedirects($options = ['limit' => self::REQ, 'min-hits' => 50]) {
+    $query = $this->database->select('node__field_legacy_redirects_legacyurl', 'source')
+      ->fields('source', ['entity_id', 'delta']);
+    $query->addField('source', 'field_legacy_redirects_legacyurl_value', 'source');
+    $query->addField('target', 'field_legacy_redirects_ref_conte_target_id', 'target');
+    if ($options['limit']) {
+      $query->range(0, $options['limit']);
```

Comment:
Um, there's no pagination happening here AFAICT.  I think we either want to execute this query unlimited and just rely on PHP's efficiency in buffering records (which should be ok, based on past experience), or actually do pagination.

Reply:
Agreed. The --limit is there just for development purposes. I've added a note to that effect in the Help for the command.